uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Import/Parser/Base.php

```diff
@@ -500,11 +503,78 @@ private function createLeaveRequestFromImportData($params) {
       $payload['sickness_reason'] = $sicknessReasons['other'];
     }
 
-    $params['request_type'] = $payload['request_type'];
-    $this->validateLeaveParams($params);
+    $this->validateLeaveParams(array_merge($params, $payload));
+    $payload = array_merge($payload, $this->getAdditionalPayload($params['absence_type']));
     return LeaveRequest::create($payload, LeaveRequest::IMPORT_VALIDATION);
   }
 
+  /**
+   * Sets the time for the from_date and to_date of a leave
+   * request whose balance change is to be calculated in days.
+   * It sets the time of the from_date as '00:00' and the
+   * time for the to_date as '23:59'
+   *
+   * @param array $params
+   * @param string $absenceType
+   */
+  private function setTimeForLeaveDates(&$params, $absenceType) {
+    if(!$this->isCalculationUnitInDays($absenceType)) {
+      return;
+    }
+
+    $fromDate = new DateTime($params['from_date']);
+    $fromDate->setTime(00, 00);
+    $toDate = new DateTime($params['to_date']);
+    $toDate->setTime(23, 59);
+
+    $params['from_date'] = $fromDate->format('YmdHis');
+    $params['to_date'] = $toDate->format('YmdHis');
+  }
+
+  /**
+   * Checks whether the absence type calculation unit is in days
+   * or not.
+   *
+   * @param string $absenceType
+   *
+   * @return bool
+   */
+  private function isCalculationUnitInDays($absenceType) {
+    $calculationUnit = $this->absenceTypes[$absenceType]['calculation_unit'];
+
+    return $this->calculationUnitOptions['days'] == $calculationUnit;
```

Comment:
If the absence types were objects, you could kept the logic encapsulated in the Absence Type BAO and this method could be something more like this:
```php
return !$this->absenceTypes[$absenceType]->isCalculationUnitInHours();
```

Reply:
Good Idea.